sovrin_client/agent/agent.py

```diff
@@ -190,9 +196,18 @@ def __init__(self,
                  attrRepo=None):
 
         Agent.__init__(self, name, basedirpath, client, port, loop=loop)
-        self._wallet = wallet or Wallet(name)
+
+        self.config = getConfig(basedirpath)
+
+        self._restoreLastActiveWallet()
+
+        if wallet:
+            self._wallet = wallet
```

Comment:
This is already there, it is not introduced change, and I am not sure why it was made private from the beginning. So, for now, I'll keep it as it is, but I'll see if I can get some information on this and later on change it.